### ISSUE TITLE
Mumble lags uncontrollably when playing L4D2 

### ISSUE THREAD

**Issue by Kevin-Schnaubelt-Jr** [mumble-voip/mumble#5728]
### The issue

So I have two computers. Only one is having this issue. If I have the source engine plugin enabled, the mumble app with slow to a crawl. I almost can't click on anything. If i try to talk, the other computer basically can't hear me at all because Mumble isn't able to pick up my voice. 

The other computer has no issues whatsoever. Settings in mumble are the same across two computers. 

### Mumble version

1.4.0

### Mumble component

I don't know

### OS

Windows

**Comment by davidebeatrici** [comment]
Is there something related in the developer console?

Also, did you mean 1.4.230?

**Comment by Wayne-The-Brain-McClain** [comment]
I'm on version 1.4.230 yes, sorry. And I'm not sure.

**Comment by Wayne-The-Brain-McClain** [comment]
Tried it for L4D1 as well and same issue. If the plugin is enabled, the Mumble application is hitching and lagging.

**Comment by Wayne-The-Brain-McClain** [comment]
Okay so update. "Link to game and transmit position" was not checked on the computer that wasn't lagging. I checked it and now Mumble on both computers lag uncontrollably when a source engine game is running. 

I could have sworn it was actually picking up my audio on the one computer. Going to do further testing.

**Comment by davidebeatrici** [comment]
Was it picking up your audio output as input?

**Comment by Wayne-The-Brain-McClain** [comment]
I'm not sure what it was picking up. 

Just tried to play with it, and the app is so laggy that it only ever transmit a quick millisecond of sound every couple of seconds. 

I don't have to be in a server or anything though. As long as the game is running in the background, my audio is hitchy and laggy. Even in the audio tuning wizard.

**Comment by Wayne-The-Brain-McClain** [comment]
I've tried the Borderlands 2 plugin on both computers and the application doesn't lag there. Going to try and actually start a coop game and see if the actual positional audio works.

**Comment by vimpostor** [comment]
Does it display a message that the Source Engine positional audio plugin is linked in the left log column?

Would be interesting to see if the problem is because it is failing to link and thus trying to link repeatedly or if the performance problem is in the retreiving of positional audio when already linked.

**Comment by vimpostor** [comment]
> Any suggestions?

You could start by answering my [still unanswered question](https://github.com/mumble-voip/mumble/issues/5728#issuecomment-1166463399).

**Comment by DkBobo** [comment]
> > Any suggestions?
> 
> You could start by answering my still unanswered [question](https://github.com/mumble-voip/mumble/issues/5728#issuecomment-1166463399).

I don't know as this is my first time using mumble.  I didn't see any messages but i'm not quite sure what the log column is.

**Comment by Krzmbrzl** [comment]
The log column is the same as the chat window. Did you see anything that claimed that the plugin has linked to the game?

**Comment by DkBobo** [comment]
> The log column is the same as the chat window. Did you see anything that claimed that the plugin has linked to the game?

No there wasn't anything. I'll try running it as Administrator next time maybe

**Comment by Krzmbrzl** [comment]
Okay then this would indicate that thevlag is due to the plugin trying to link :thinking:

@davidebeatrici do you have any idea what might be causing this?

**Comment by DkBobo** [comment]
> Okay then this would indicate that thevlag is due to the plugin trying to link 🤔
> 
> @davidebeatrici do you have any idea what might be causing this?

I tried playing Borderlands 2 and there wasn't a message either. Can you link a tutorial for how to link any game to mumble because all I can find are tutorials for Minecraft.

**Comment by DkBobo** [comment]
To my understanding, all you need to do is tick the "Link game to game and transmit position" on and tick "enable" and "PA" for Source Engine (or any other plugins depending on the game). I've also ticked off the enable box for the "Link" plugin. Then you enter any server and the game you want to play and there should be a message saying the game has linked (?)

**Comment by davidebeatrici** [comment]
Exactly. Please note that most plugins are not guaranteed to work with the latest game versions.

The Link plugin exposes a shared memory area that games can write to, it's the most reliable method but requires support from the developers.

The Source Engine plugin uses pattern matching to find specific assembly code in memory, there is a chance something has been changed in the game. I will take a look.

**Comment by davidebeatrici** [comment]
I was able to reproduce the issue, it appears to be Windows-specific.

For reference:

```
Version 2.2.2.9 (left4dead2)
Network Version 2.1.0.0
Exe build: 12:56:10 Apr 24 2023 (8934) (550)
```

The plugin works flawlessly with the exact same version of the game running on Linux through Proton.
